api/task.go

```diff
@@ -184,14 +185,22 @@ func (h *taskHandler) updateTask(w http.ResponseWriter, r *http.Request) {
 }
 
 type TaskLifeCycleHandler struct {
-	// TODO: replace this when implementing endpoint
-	//tasks map[string]oapigen.Task
-	//lock  sync.Mutex
+	mu *sync.RWMutex
+
+	store   *event.Store
+	drivers *driver.Drivers
+	version string
+	tasks   map[string]oapigen.Task
 }
 
-func NewTaskLifeCycleHandler() *TaskLifeCycleHandler {
+func NewTaskLifeCycleHandler(store *event.Store, drivers *driver.Drivers,
+	version string) *TaskLifeCycleHandler {
 	return &TaskLifeCycleHandler{
-		//tasks: make(map[string]oapigen.Task),
+		mu:      &sync.RWMutex{},
+		store:   store,
+		drivers: drivers,
+		version: version,
+		tasks:   make(map[string]oapigen.Task),
```

Comment:
you can leave this line commented out, it's only for testing

Reply:
Oh, okay! I should have asked about what it was for but I forgot! Will we need this tasks map in the future beyond for testing?

Reply:
No I don't think it'll be needed. It was just so that I could test creating a task in an in memory store so I could then delete it using the delete endpoint. Just for some added "realism" to the testing scenario.